Sources/iOSClientExposureDownload/Download/SessionManager+Exposure.swift

```diff
@@ -440,26 +441,177 @@ extension iOSClientDownload.SessionManager where T == ExposureDownloadTask {
     }
     
     
-    /// Check if the license has expired or not
+    /// Check if the downloaded Asset has expired or not
     /// - Parameter assetId: asset id
     /// - Returns: true if the license has expired
-    public func isExpired(assetId: String)-> Bool {
+    public func isExpired(assetId: String, environment: Environment, sessionToken: SessionToken, completionHandler: @escaping (Bool) -> Void) {
         let downloadedAsset = getDownloadedAsset(assetId: assetId)
-        guard let playTokenExpiration = downloadedAsset?.entitlement?.playTokenExpiration else {
-            return true
+        
+        if self.isConnectedToNetwork() {
+            // Internet connection found
+            
+            // Get download verified information
+            self.getDownloadVerified(assetId: assetId, environment: environment, sessionToken: sessionToken) { [weak self] verifiedInfo in
```

Comment:
what if network is connected, but this request to `self.getDownloadVerified` fails?

Reply:
I passed it as an expired downloaded asset , `Calculation failed or did not return , assume downloaded asset is expired` , but perhaps that may be bit harsh

Reply:
I think that is misleading to call it as "expired". It's better to throw an error that indicates the request failed due to server or network issue, and let the app decide what to do.

Reply:
Noted , I will fix it